compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/intrinsics/ArrayIterator.kt

```diff
@@ -29,11 +31,15 @@ object ArrayIterator : IntrinsicMethod() {
         signature: JvmMethodSignature,
         context: JvmBackendContext
     ): IrIntrinsicFunction {
-        val owner = context.typeMapper.mapClass(expression.symbol.owner.parentAsClass)
+        val isPrimitive = ((expression.type as? IrSimpleType)?.arguments?.size ?: 0) == 0
+        val owner =
+            if (isPrimitive) context.typeMapper.mapClass(expression.symbol.owner.parentAsClass)
+            else context.typeMapper.mapClass(context.irBuiltIns.arrayClass.owner)
+
```

Comment:
Could you please clarify why these changes in the backend are necessary? Right now this intrinsic works correctly in the old frontend mode. Is there some difference in the IR here? In that case I believe it's best to fix it in fir2ir.

Reply:
I'll look for a fix in fir2ir and amend the PR.

Reply:
Restored this file.